### ISSUE TITLE
Docs on adding extensions is misleading or incomplete

### ISSUE THREAD

**Issue by alexgleith** [stac-utils/pystac#602]
The [Extensions docs](https://pystac.readthedocs.io/en/latest/concepts.html#extensions) say that to add an extension you do:

```python
import pystac
from pystac.extensions.eo import EOExtension

item = Item(...)  # See docs for creating an Item
eo_ext = EOExtension.ext(item)
```

But in practice you need to do:

```python
EOExtension.add_to(item)
eo = EOExtension.ext(item)
```

Is this intended, and if yes, can we tweak the docs please?

**Comment by duckontheweb** [comment]
Thank for raising this @alexgleith. 

The behavior of raising an exception if the Item does not already contain the schema URI is intended (see [this PR comment](https://github.com/stac-utils/stactools/pull/113#discussion_r638974407) in `stactools` for some background).

Those docs assumed that the Item already has the EO extension schema URI in its `stac_extensions` list, but we should definitely make it more clear that you need to use either `EOExtension.add_to(item)` or `eo_ext = EOExtension.ext(item, add_if_missing=True)` to add the extensions to an Item that does not already implement it.

**Comment by alexgleith** [comment]
Is there a good reason why that doesn't default to true?

**Comment by gadomski** [comment]
https://github.com/stac-utils/stactools/pull/113#discussion_r639860502 has some discussion about why it's not automatic, but basically we wanted to force folks to be explicit about adding any new extension schema urls so we didn't, e.g., mutate the object in question (by adding an extension schema url) when a person was treating the object as "read-only".

**Comment by alexgleith** [comment]
Hmm, I can see the logic here for when dealing with an existing item.

I reckon the docs should cover it better still. Maybe I need to read them again to see if I missed it somewhere!

**Comment by duckontheweb** [comment]
@alexgleith I opened #701 to try to clarify this better in the Concepts docs. Let me know if you think that covers it.

**Comment by duckontheweb** [comment]
Closed via #701
